src/server.lisp

```diff
@@ -147,123 +147,132 @@ By default, a symbol passed in for F will be automatically converted into the na
   |error|)
 
 
+(defun log-completion-message (logger request reply start-time)
+  (macrolet ((log-it (priority control &rest args)
+               `(cl-syslog:rfc-log
+                    (logger ,priority ,control ,@args)
+                  (:msgid "LOG0002")
+                  (|rigetti@0000|
+                   |methodName| (|RPCRequest-method| request)
+                   |requestID| (|RPCRequest-id| request)
+                   |wallTime| (format nil "~f" (/ (- (get-internal-real-time) start-time)
+                                                  internal-time-units-per-second))
+                   |error| ,(if (<= (cl-syslog:get-priority priority)
+                                    (cl-syslog:get-priority ':err))
+                                "true"
+                                "false")))))
+    (etypecase reply
+      (|RPCReply|
+       (log-it :info "Requested ~a completed" (|RPCRequest-method| request)))
+      (|RPCError|
+       (log-it :err "Request ~a error: ~a" (|RPCRequest-id| request) (|RPCError-error| reply))))))
+
+
+(defvar *warnings*)
```

Comment:
If we're ok with ignoring warnings that happen during `%pull-raw-request` or deserialization, then `*warnings*` could go back to being a regular lexical var in the body of `%process-request`.

Reply:
that was my thinking as well